examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,308 @@
+	_, error := os.Open(fPath)
+	return !errors.Is(error, os.ErrNotExist)
+}
+
+func WriteKeysToSSHDir(seckeySshPem []byte, certBytes []byte) error {
+	homePath, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	sshPath := filepath.Join(homePath, ".ssh")
+
+	//  To enable ssh to automatically find the key created by openpubkey we
+	// need to use a default ssh key path. However this means that this
+	// filename might already be in use by the user. To ensure we don't
+	// overwrite a ssh key not created by openpubkey we check the comment in the
+	// key to see if it was created by openpubkey
+	defaultKeyNames := []string{"id_ecdsa_sk", "id_ecdsa", "id_dsa"}
+	for i := range []string{"id_ecdsa_sk", "id_ecdsa", "id_dsa"} {
+		seckeyPath := filepath.Join(sshPath, defaultKeyNames[i])
+		pubkeyPath := seckeyPath + ".pub"
+
+		if !FileExists(seckeyPath) {
+			// If ssh key file does not currently exist, we don't have to worry about overwriting it
+			return WriteKeys(seckeyPath, pubkeyPath, seckeySshPem, certBytes)
+		} else if !FileExists(pubkeyPath) {
+			continue
+		} else {
+			// If ssh key does file does exist, check if it is an openpubkey file, if it is then it is safe to overwrite
+			sshPubkey, err := os.ReadFile(pubkeyPath)
+			if err != nil {
+				fmt.Println("Failed to read:", pubkeyPath)
+				continue
+			}
+			sshPubkeyTok := strings.Split(string(sshPubkey), " ")
+			if len(sshPubkeyTok) != 3 {
+				fmt.Println("Failed to parse:", pubkeyPath)
+				continue
+			}
+			// check if pubkey comment to see if it an openpubkey ssh key
+			if strings.Contains(sshPubkeyTok[2], ("openpubkey")) {
+				// safe to overwrite
+				return WriteKeys(seckeyPath, pubkeyPath, seckeySshPem, certBytes)
+			}
+		}
+	}
+	return fmt.Errorf("no default ssh key file free for openpubkey")
+}
+
+func log(line string) {
+	f, err := os.OpenFile("/var/log/openpubkey.log", os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0700)
+	if err != nil {
+		fmt.Println("Couldn't write to file")
+	} else {
+		defer f.Close()
+		if _, err = f.WriteString(line + "\n"); err != nil {
+			fmt.Println("Couldn't write to file")
+		}
+	}
+}
+
+// This code is currently intended as an example for how OpenPubkey can secure SSH access.
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Printf("Example SSH key generator using OpenPubkey: command choices are: login, ver")
+		return
+	}
+	command := os.Args[1]
+
+	op := parties.GoogleOp{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Issuer:       issuer,
+		Scopes:       scopes,
+		RedirURIPort: redirURIPort,
+		CallbackPath: callbackPath,
+		RedirectURI:  redirectURI,
+	}
+
+	switch command {
+	case "login":
+		{
+			if len(os.Args) != 2 {
+				fmt.Println("Invalid number of arguments for login, should be `opkssh login`")
+				os.Exit(1)
+			}
+
+			// If principals is empty the server does not enforce any principal.
+			// The OPK verifier should use policy to make this decision.
+			principals := []string{}
+
+			gqFalse := false
```

Comment:
I don't like just passing `false` in as a parameter because it is hard to tell what is being set to false.

Ideally I would like to do this:
```
client := &parties.OpkClient{
				Op: &op,
				GqSig: false.
			}
```
clear and optimal number of lines

Reply:
but you do know what it's referring to because of the functional signature although I guess it's called `gqFlag` which doesn't actually clear it up. That's fine to keep this variable this name, hopefully we'll change it down the line